open_feature/open_feature_api.py

```diff
@@ -4,10 +4,12 @@
 from open_feature.open_feature_client import OpenFeatureClient
 from open_feature.provider.provider import AbstractProvider
 
-_provider = None
+_provider: typing.Optional[AbstractProvider] = None
```

Comment:
This isn't really a `typing` problem, so no big deal if you don't want to address it in this PR, but the provider should probably default to the `noop_provider`. Technically, it doesn't _have to_, but the API must no-op if no provider is registered and defaulting to the no-op is generally a good way to do it.

Apologies if this is handled elsewhere.

Reply:
I agree, then the provider can always be defined 👍

Reply:
I think it would be good for us to move forward with this since we have a number of typing improvements we'll receive here, and this isn't actually changing the current code, just adding typing.